### ISSUE TITLE
Do you have to append unsafe-inline?

### ISSUE THREAD

**Issue by will** [twitter/secure_headers#403]
I just tried to switch from 'self' and domains to 'strict-dynamic' and nonces for the tags, however when doing that, this code 

``` ruby
    # Private: adds a nonce or 'unsafe-inline' depending on browser support.
    # If a nonce is populated, inline content is assumed.
    #
    # While CSP is backward compatible in that a policy with a nonce will ignore
    # unsafe-inline, this is more concise.
    def append_nonce(source_list, nonce)
      if nonce
        source_list.push("'nonce-#{nonce}'", UNSAFE_INLINE)
      end

      source_list
    end
```
https://github.com/twitter/secure_headers/blob/647dcc4c4dd281f0a0744e64da917dbe298994d3/lib/secure_headers/headers/content_security_policy.rb#L180

appended 'unsafe-inline'. I get that it works and the browsers will ignore the unsafe-inline, but I'd be more comfortable if it wasn't there at all. Is there any way to avoid this?

**Comment by oreoshake** [comment]
It would require a code change. I'm not opposed to making that change but it's there for good reason and building an opt-out option would be a good idea.

**Comment by will** [comment]
I'm assuming that the good reason is for backwards compatibility? If that is the case, for my particular site, I'm fine forcing all of my customers to be on up-to-date browsers.

**Comment by oreoshake** [comment]
> I'm assuming that the good reason is for backwards compatibility?

Correct, any clients that don't understand the new directive source will fallback to the `'unsafe-inline`' and those that do understand it, ignore the `'unsafe-inline'`.

**Comment by will** [comment]
For anyone finding this issue in the future, it has been implemented in #404 .

**Comment by oreoshake** [comment]
I'm trying to push a new gem but I think rubygems is having issues? I'll try again later.

```html
<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
 "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html>
  <head>
    <title>502 Bad Gateway</title>
  </head>
  <body>
    <h1>Error 502 Bad Gateway</h1>
    <p>Bad Gateway</p>
    <h3>Guru Mediation:</h3>
    <p>Details: cache-bur17539-BUR 1561580744 879826524</p>
    <hr>
    <p>Varnish cache server</p>
  </body>
</html>
```
